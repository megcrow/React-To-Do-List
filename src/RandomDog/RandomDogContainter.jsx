import { compose, withState, withHandlers, lifecycle } from 'recompose';

import RandomDog from './RandomDog';

const enhance = compose(
    withState('apiResponse', 'setApiResponse', {
        response: "",
        loading: false
    }),
    withState('dogUrl', 'setDogUrl', ''),
    withState('isLoadingDog', 'setIsLoadingDog', false),

    withHandlers({
        onGetRandomDog: ({ setDogUrl, setIsLoadingDog, setApiResponse, apiResponse }) => async () => {
            try {
                setIsLoadingDog(true)
                const response = await fetch('https://dog.ceo/api/breeds/image/random');
                const responseJSON = await response.json();
                const message = responseJSON.message;
                setDogUrl(message);
                setIsLoadingDog(false);
            } catch(error) {
                setIsLoadingDog(false)
                console.log('error', error);
            }
        }
    }),

    lifecycle({
        componentDidMount() {
            this.props.onGetRandomDog();
        }
    })
)
export default enhance(RandomDog)
