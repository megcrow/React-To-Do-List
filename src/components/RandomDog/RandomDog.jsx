import React from 'react';

import './RandomDog.css'

export default function RandomDog ({ dogUrl, isLoadingDog }){
    return(
        <div>
        {
            isLoadingDog ? (
                <p className="loading-text">Loading...</p>
            ) : (
                <img src={dogUrl} alt="header" className="header-img" />
            )
        }
        </div>
    )
}
