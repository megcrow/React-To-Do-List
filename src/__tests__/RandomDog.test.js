import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme,{ shallow } from 'enzyme';

import RandomDog from '../components/RandomDog/RandomDog';
import mockImage from '../__mockData__/mockImage.jpg'

Enzyme.configure({ adapter: new Adapter() })

describe('RandomDog', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <RandomDog />
        );
    });

    test('RandomDog component should match snapshot', () => {
        expect('wrapper').toMatchSnapshot();
    });

    test('should say `Loading...` when loading is true', () => {
        wrapper.setProps({isLoadingDog: true})
        expect(
            wrapper.find('.loading-text').text()
        ).toBe('Loading...');
    });

    test('should show image when loading is false', () => {
        wrapper.setProps({dogUrl: mockImage, isLoadingDog: false})
        expect(
            wrapper.contains(<img src={mockImage} alt="header" className="header-img"/>)
        ).toBe(true);
    });
})
