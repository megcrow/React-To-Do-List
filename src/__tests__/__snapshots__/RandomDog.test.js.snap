// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomDog RandomDog component should match snapshot 1`] = `"wrapper"`;
